CMakeLists.txt

```diff
@@ -55,12 +55,12 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 # Options for building certain parts of the repo. Everything is built by default.
 option(BUILD_SHARED_LIBS "Build with shared libs (needed for JNI)" ON)
-option(WITH_JAVA "Include Java and JNI in the build" ON)
+option(WITH_JAVA "Include Java and JNI in the build" OFF)
 option(WITH_JAVA_SOURCE "Build Java source jars" ${WITH_JAVA})
 option(WITH_CSCORE "Build cscore (needs OpenCV)" ON)
 option(WITH_NTCORE "Build ntcore" ON)
 option(WITH_WPIMATH "Build wpimath" ON)
-option(WITH_WPIUNITS "Build wpiunits" ON)
+option(WITH_WPIUNITS "Build wpiunits" ${WITH_JAVA})
```

Comment:
this would be a good place to use `cmake_dependent_option` to make sure that `WITH_WPIUNITS` can only be enabled alongside `WITH_JAVA`

Reply:
wpiunits has always supported a standalone build. Do we want to remove that? Admittedly, I don't think many people would use that...

Reply:
It doesn't actually build anything if WITH_JAVA isn't enabled.

Reply:
Should `WITH_JAVA_SOURCE` be changed to also use `cmake_dependent_option`?

Reply:
Maybe? I purposefully kept the ability to build Java source JARs without needing to build the actual JARs, but use is probably limited. I'm leaning towards keeping it though.